pyracf/user/user_admin.py

```diff
@@ -275,24 +275,31 @@ def take_away_auditor_authority(self, userid: str) -> Union[dict, bytes]:
     # Password
     # ============================================================================
     def set_password(
```

Comment:
This might be a little cleaner, but I am not sure how a password set to `False` is intended to affect the logic here.

```python
def set_password(
Leave a comment
        self, userid: str, password: Union[str, bool], expired: bool = True
    ) -> Union[dict, bytes]:
        """Set a user's password."""
        traits = {"base:password": password}
        if expired:
            traits["base:password_expired"] = True
        return self.to_steps(self.alter(userid, traits=traits))
```

Reply:
Change `if password:` to `if expired` and rename `request_traits` to `traits` since this is consistent with naming used elsewhere. Do the same with `set_passphrase`.

Reply:
So, you are saying that if `expired` is `False`, we want to explicitly set `base:password_expired` to `False`, otherwise we don't specify anything for `base:password_expired` which causes RACF's default behavior to take effect *(set the new password as expired by default)*?

Reply:
Either way, I also think there should be tests for both `expired=True` and `expired=False`. The test for `expired=True` is a little redundant, but I think it would be a good regression test. If the default behavior differs from setting `expired=True` explicitly, then that is indicative of a problem.

Reply:
That's a fair point. I changed things around a bit again. This is the code logic. As long as expired is specified explicitly, the value will be used. If it is not, it won't be. This should be functionally the same as it was, but this will be the cleanest way to deliver exactly what the user would want as a result.

```python3
    # ============================================================================
    # Password
    # ============================================================================
    def set_password(
        self, userid: str, password: Union[str, bool], expired: Union[bool, None] = None
    ) -> Union[dict, bytes]:
        """Set a user's password."""
        traits = {"base:password": password}
        if expired is not None:
            traits["base:password_expired"] = expired
        result = self.alter(userid, traits=traits)
        return self._to_steps(result)

    # ============================================================================
    # Passphrase
    # ============================================================================
    def set_passphrase(
        self, userid: str, passphrase: Union[str, bool], expired: Union[bool, None] = None
    ) -> Union[dict, bytes]:
        """Set a user's passphrase."""
        traits = {"base:passphrase": passphrase}
        if expired is not None:
            traits["base:password_expired"] = expired
        result = self.alter(userid, traits=traits)
        return self._to_steps(result)
```

I then added explicit test cases for password and password phrase with expired=True and the additional request xml files to match.